File: PS-admin/Reservation.cpp
#include "Reservation.h"

Reservation::Reservation(const int ID_reservation, const string date_debut, const string date_fin,const string statut, const string nom_locataire)
{
    ID_Reservation=ID_reservation;
    Date_Debut=date_debut;
    Date_Fin=date_fin;
    Statut=statut;
    Nom_Locataire=nom_locataire;

}

Reservation::~Reservation()
{

}

void Reservation::Afficher_Reservation() const
{
    cout<<"de: "<<Date_Debut<<"  a: "<<Date_Fin<<", par :"<<Nom_Locataire<<endl;
}

void Reservation::Set_Annonce(Annonce* A)
{
    Annonce_Reservation=A;
}

File: PS_hote/reservation.java
import java.util.Date;

public class reservation {
    private int id;
    private Date date_debut;
    private Date date_fin;
    private double commssion_platform;
    private double montant_Hote;
    private String NomLocataire;

    private annonce annonce;
    private locataire locataire;

    // Constructor
    public reservation(int id, Date date_debut, Date date_fin, annonce annonce, locataire locataire) {
        this.id = id;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.annonce = annonce;
        this.locataire = locataire;
        // Calculate total based on number of days and price per night
        long diffInMillies = Math.abs(date_fin.getTime() - date_debut.getTime());
        long diffInDays = diffInMillies / (24 * 60 * 60 * 1000);
//        this.total = annonce.getPrix() * diffInDays;
    }
    public reservation(int id, Date date_debut, Date date_fin,String NomLocataire,double montant_Hote) {
        this.id = id;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.NomLocataire= NomLocataire;
        this.montant_Hote=montant_Hote;
        
    }

    // Methods
    public void confirmer() {
        System.out.println("Réservation confirmée pour: " + this.annonce.getTitre());
    }

    public void annuler() {
        System.out.println("Réservation annulée pour: " + this.annonce.getTitre());
    }
    
    public void afficherReservation() {
        System.out.println("de: " + this.date_debut + "  a: " + this.date_fin + ", par: " 
        		+ this.NomLocataire+" ,d'un montant :"+(this.montant_Hote-(this.montant_Hote*0.03)) + " DH");
    }

}
